### ISSUE TITLE
Add multicolor feature to line and area

### ISSUE THREAD

**Issue by stsdc** [lcallarec/live-chart#34]
It would be super cool to be able change colour of segment of the line to indicate e.g. high values. That also would be nice to have for the area under such segment.

![image](https://user-images.githubusercontent.com/6031763/158023939-1aad8afa-40d0-4509-92f4-49b8963909fc.png)

**Comment by lcallarec** [comment]
Ok, if you need that feature, I'll take a look.

**Comment by stsdc** [comment]
ping :eyes:

**Comment by lcallarec** [comment]
Shame on me. Time flies. I promise to take a look this week.

**Comment by lcallarec** [comment]
Just a small update to keep you informed : I'm working on a poc. I should be able to release this feature next week, + this one https://github.com/lcallarec/live-chart/issues/36

**Comment by lcallarec** [comment]
Just asking : do you have any needs to define more than one "area", i.e. "high (in orange)" and "very high (in red)" ?

**Comment by stsdc** [comment]
Not really, as for now 😅
Can't really think right now for other use cases for Monitor. Hmm maybe in profiling? As we can see in browser web tools. But I'm not going to do anything like that in the near future.

**Comment by lcallarec** [comment]
Ok, that's a good news :) Having more that one area is a pain ^^.

I'll release a beta before the end of the week, for SmoothLineArea only.

That feature wasn't so obvious to code and could have some impacts on performances and stability.
It'll better if you could take some times to let this beta running inside your Monitor application a long time, just to check if everything is ok.

**Comment by lcallarec** [comment]
To test this feature, please update to 55d904b56bcb920e25a3f459572e48e99a92aeaf.

Usage :
```vala
var red = Gdk.Color() { red: 1, green: 0, blue: 0, alpha: 1};
var lightly_red = Gdk.Color() { red: 1, green: 0, blue: 0, alpha: 0.5};

var line = new LiveChart.SmoothLineArea(); //or SmoothLine();

line.region = new LiveChart.Region.between(100, double.MAX).with_line_color(red).with_area_color(lightly_red);
```
I'm waiting for your feedbacks !

**Comment by stsdc** [comment]
Looks cool! (temperature)

But experienced a few crushes, maybe because I tried to colour area when using stacked charts on main CPU chart. Added `if (series_quantity == 1)` and it's kinda stable.

Thanks!

![Zrzut ekranu z 2022-11-01 21 18 05](https://user-images.githubusercontent.com/6031763/199332948-74903a2c-f670-4b1d-8bab-d38bbf3eb763.png)

You can check here https://github.com/stsdc/monitor/blob/6a6a4dbe8de14e89de4bcbe9d9c589f56af4220b/src/Widgets/Chart/Chart.vala#L49-L68

PS I believe you meant `Gdk.RGBA` and `=` :sweat_smile:

**Comment by lcallarec** [comment]
I'm a bit frightened about the crash you experienced. Was it a segfault ? Have you got any log about that crash ?

**Comment by stsdc** [comment]
Here is a gdb output, hope it helps:

```
Thread 1 "com.github.stsd" received signal SIGSEGV, Segmentation fault.
0x00007ffff7f9ea21 in live_chart_region_handle (Python Exception <class 'ValueError'> Variable 'static_fundamental_type_nodes' not found.: 
self=, intersections=0x7fffe8027ed0, previous=0x7fffffffad20, current=0x7fffffffad50, 
    get_intersection=0x7ffff7f94c4b <___lambda12__live_chart_get_intersection>, get_intersection_target=0x7fffdc023460)
    at ../subprojects/live-chart/src/region.vala:82
82                              intersections.close(coords.x, exited_at);
```

**Comment by lcallarec** [comment]
Yes, thanks, it helps. Let me figure out what happened.

**Comment by lcallarec** [comment]
@stsdc : would you please test monitor with this commit of live-chart : https://github.com/lcallarec/live-chart/commit/1c090e18fa4ae9f3ccfe0f38d619ebeccd7d6361

Should fix the issue you noticed, which was - I believe - unrelated to the number of series you had on the chart.

Let me know if it's working.

**Comment by stsdc** [comment]
Can confirm everything works! It was probably just a strange case were only one thread was loaded plus I have a pretty hi dpi, so wasn't able to notice. Sorry for that :sweat_smile:  and thanks for this feature. Issue can be closed!

![Zrzut ekranu z 2022-12-10 18 14 32](https://user-images.githubusercontent.com/6031763/206867055-fcbb4c04-b4d6-4526-baf8-2973d41df517.png)

**Comment by lcallarec** [comment]
Finally released in 1.10.0.
